File: commands/commit-commands/add.js
import simpleGit from "simple-git";

const git = simpleGit();

export default {
  command: "add [files]",
  describe: "Add files to the Git index (staging area)",
  builder: (yargs) => {
    yargs.positional("files", {
      describe: "Files to add (use '.' to add all files)",
      type: "string",
      demandOption: true,
    });
  },
  handler: async (argv) => {
    try {
      await git.add(argv.files);
      console.log(`Files added: ${argv.files}`);
    } catch (error) {
      console.error("Error adding files:", error.message);
    }
  },
};

File: cmd/add.go
package cmd

import (
	"fmt"
	"log"
	"os/exec"

	"github.com/spf13/cobra"
)

var AddCmd = &cobra.Command{
	Use: "add",
	Short: "Add files to the staging area",
	Run: func(cmd *cobra.Command, args []string) {
		if len(args) < 1 {
			fmt.Println("Error: You must provide at least one file to add")
			return
		}

		for _, file := range args {
			err := exec.Command("git", "add", file).Run()
			if err != nil {
				log.Fatalf("Error adding file '%s': %v", file, err)
			}
			fmt.Printf("File '%s' added to staging area\n", file)
		}
	},
}

func init() {
	//No flags
}